common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -136,7 +136,11 @@
   private DB db;
 
   public YarnShuffleService() {
-    super("spark_shuffle");
+    this("spark_shuffle");
```

Comment:
It _is_ still hardcoded. I haven't found a way to access Spark configuration from that constructor and `org.apache.hadoop.yarn.server.api.AuxiliaryService` requires the name. Do you have a suggestion of how that could be done?

Reply:
As I commented below https://github.com/apache/spark/pull/26000#discussion_r330742783, if this is just for yarn, put it in YarnShuffleService?

Reply:
It is hardcoded here. Once the shuffle service name is configured, won't they mismatch? Will it cause problem?

Reply:
It *is* hardcoded *here*. HDP hardcodes another value though (`spark2_shuffle`). While vanilla Spark would keep working as is and would use the name `spark_shuffle`, the new configuration option would allow users to point Spark to non-vanilla shuffle service.
The changes to that class are done only to test that changing the name of the service and in the configuration play nicely together.

Reply:
It seems impossible to register the service with the name passed in the configuration because the configuration is passed *after* the class is instantiated.

Reply:
I see. So this config can only be used to let Spark choose which service to connect. It cannot change the name of Shuffle Service.

Reply:
Yes. I guess I could implement a workaround, which would get the config setting from the default `Configuration`, but that, at least theoretically, wouldn't guarantee that the exact configuration would be passed during service initialization.